src/cocaine-app/yt_worker.py

```diff
@@ -0,0 +1,176 @@
+from logging import getLogger
+from yql.api.v1.client import YqlClient
+import time
+import datetime
+
+
+logger = getLogger('mm.planner')
+
+
+class YqlWrapper(object):
+    def __init__(self, cluster, token, attempts, delay):
+        self._cluster = cluster
+        self._token = token
+        self._attempts = attempts
+        self._delay = delay
+        self.aggregate_query = """
+        SELECT couple_id
+        FROM
+            (SELECT
+                        couple_id,
+                        SUM(expired_size) AS sum_expired_size
+                    FROM [{table}]
+                    WHERE expiration_date <= {timestamp}
+                    GROUP BY couple_id)
+            WHERE  sum_expired_size >= {trigger};
+        """
+
+        # XXX: maybe something more optimal could be done? How the db is organized?
+        # XXX: tounixdate -> arithmetic ops
+        # XXX: use HAVING ?
+        self.preaggregate_query = """
+$dateconv = @@
+import time
+import datetime
+def tounixdate(exp_time):
+    return int(time.mktime((datetime.date.fromtimestamp(exp_time) + datetime.timedelta(days=1)).timetuple()))
+@@;
+
+$dateconvert = Python::tounixdate("(Uint64?)->Uint64", $dateconv);
+
+INSERT INTO [{agr_table}]
+SELECT
+    couple_id,
+    expiration_date,
+    namespace,
+    "upload" AS operation,
+    SUM(CAST(object_size AS Int64)) AS expired_size,
+    "{date_iso}" AS source_table,
+    {timestamp} AS timestamp
+FROM [{main_table}]
+WHERE op="upload"
+GROUP BY
+    couple_id,
+    namespace,
+    $dateconvert(CAST(expire_at AS Uint64)) AS expiration_date;
+
+INSERT INTO [{agr_table}]
+SELECT
+    couple_id,
+    expiration_date,
+    namespace,
+    "delete" AS operation,
+    -1*SUM(CAST(object_size AS Int64)) AS expired_size,
+    "{date_iso}" AS source_table,
+    {timestamp} AS timestamp
+FROM [{main_table}]
+WHERE op="delete" AND expire_at IS NOT NULL
+GROUP BY
+    couple_id,
+    namespace,
+    $dateconvert(CAST(expire_at AS Uint64)) AS expiration_date;
+"""
+
+    def send_request(self, query, timeout=None):
+        """
+        Send request to YQL: ping verification + attempts retry
+        @return result dict on success, excepts otherwise
+        """
+        logger.debug('Send YQL request {}'.format(query))
+        result = None
+
+        with YqlClient(db=self._cluster, token=self._token) as yql:
+            try:
+                if not yql.ping():
+                    raise IOError("YQL ping failed {} {}".format(self._cluster, self._token))
+            except:
+                logger.exception("YQL ping excepted {}{}".format(self._cluster, self._token))
```

Comment:
Done in https://github.com/yandex/mastermind/pull/311/commits/4732106cd00d7b48fb9a6db30b7cff4aef8db6df